pkg/webservice/handlers.go

```diff
@@ -82,14 +82,23 @@ func getClusterInfo(w http.ResponseWriter, r *http.Request) {
 func getApplicationsInfo(w http.ResponseWriter, r *http.Request) {
 	writeHeaders(w)
 
+	queueName := r.URL.Query().Get("queue")
 	var appsDao []*dao.ApplicationDAOInfo
 	lists := gClusterInfo.ListPartitions()
 	for _, k := range lists {
 		partition := gClusterInfo.GetPartition(k)
 		appList := partition.GetApplications()
 		for _, app := range appList {
 			appDao := getApplicationJSON(app)
-			appsDao = append(appsDao, appDao)
+			if len(queueName) > 0 {
+				queueNameArr := strings.Split(app.QueueName, ".")
+				appQueueName := queueNameArr[len(queueNameArr)-1]
```

Comment:
If the app queue name is just the leaf queue than we cannot do this.
The query could be set to `root.test` and the apps runs in `root.sub.test` and your filter will return it as `test` is equal.

We should have the full queue path in the app to correctly filter and check all parts of the query parameter.